Uplift/Views/GymDetail/Facilities/CourtView.swift

```diff
@@ -0,0 +1,118 @@
+
+        courtsCollectionView = UICollectionView(frame: .zero, collectionViewLayout: flowLayout)
+        courtsCollectionView.backgroundColor = .primaryWhite
+        courtsCollectionView.isUserInteractionEnabled = false
+        courtsCollectionView.allowsSelection = false
+        courtsCollectionView.isScrollEnabled = false
+        courtsCollectionView.register(CourtCell.self, forCellWithReuseIdentifier: collectionViewIdentifier)
+
+        super.init(frame: frame)
+        backgroundColor = .primaryWhite
+        courtsCollectionView.dataSource = self
+        addSubview(courtsCollectionView)
+        setupConstraints()
+    }
+
+    required init?(coder: NSCoder) {
+        fatalError("init(coder:) has not been implemented")
+    }
+
+    private func setupConstraints() {
+        courtsCollectionView.snp.makeConstraints { make in
+            make.edges.equalToSuperview()
+        }
+    }
+
+    override func layoutSubviews() {
+        centerCollectionView()
+    }
+
+    private func update(day: Int) {
+        selectedDayIndex = day
+        let hours = facilityDetail.times.filter { $0.dayOfWeek == selectedDayIndex }
+        // Restrictions describe court
+        displayedHours = hours.flatMap { $0.timeRanges }.filter { !$0.restrictions.isEmpty }
+
+        centerCollectionView()
+        courtsCollectionView.reloadData()
+    }
+
+    // MARK: - Helper
+    /// Creates insets so collection view displays its contents in the center
+    private func centerCollectionView() {
+        let numElem = CGFloat(displayedHours.count)
+        let inset = (frame.width - (CourtView.courtSize.width * numElem)) / 2
+        courtsCollectionView.contentInset = UIEdgeInsets(top: 0, left: inset, bottom: 0, right: inset)
+    }
+
+    // MARK: - Collection View Related
+    func configure(facility: FacilityDetail, gymHours: [DailyGymHours]) {
+        facilityDetail = facility
+        dailyGymHours = gymHours
+        update(day: selectedDayIndex)
+    }
+
+    static func getHeight() -> CGFloat {
+        return courtSize.height
+    }
+
+}
+
+// MARK: - UICOllectionViewDataSource
+extension CourtView: UICollectionViewDataSource {
+
+    func collectionView(_ collectionView: UICollectionView, numberOfItemsInSection section: Int) -> Int {
+        return displayedHours.count
+    }
+
+    func collectionView(_ collectionView: UICollectionView, cellForItemAt indexPath: IndexPath) -> UICollectionViewCell {
+        guard let cell = collectionView.dequeueReusableCell(withReuseIdentifier: collectionViewIdentifier, for: indexPath) as? CourtCell else { return UICollectionViewCell() }
+        cell.configure(
+            facilityHours: displayedHours[indexPath.row],
+            dailyGymHours: dailyGymHours.filter { $0.dayOfWeek == selectedDayIndex }
+        )
+        return cell
+    }
+
+}
+
+// MARK: - Delegation
+extension CourtView: WeekDelegate {
+
+    func didChangeDay(day: WeekDay) {
+        update(day: day.index - 1)
```

Comment:
Still very confused about this -1

Reply:
its because the backend indexes mon-sun as `0..6` while `Calendar.current.component` indexes it `1..7`
I'll change it so `Weekday.index` uses the `0..6` since we tend to use that more, but that'll break other the calendar view which is being worked on atm